theano/tensor/nnet/abstract_conv.py

```diff
@@ -1751,13 +1760,114 @@ def bilinear_kernel_1D(ratio, normalize=True):
     return kern
 
 
+def frac_bilinear_upsampling(input,
+                             ratio=None,
```

Comment:
I think you can simplify this a bit by removing the `ratio` parameter.
The only current codepath that calls `frac_bilinear_upsampling` (l. 1920) is only taken if `frac_ratio` is provided and `ratio` is not.
If someone really wants to call `frac_bilinear_upsampling` manually, I think it is OK to ask for them to provide `frac_ratio=(ratio, 1)`.